libs/src/utils/fileHasher.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  ImportCandidate,
+  importer,
+  UserImporterOptions
+} from 'ipfs-unixfs-importer'
+import fs from 'fs'
+import { hrtime } from 'process'
+import { promisify } from 'util'
+import { Stream } from 'stream'
+import type { Blockstore, Options } from 'interface-blockstore'
+import type {
+  AwaitIterable,
+  Pair,
+  Batch,
+  Query,
+  KeyQuery
+} from 'interface-store'
+import type { CID } from 'multiformats/cid'
+const fsReadFile = promisify(fs.readFile)
+
+// Base functionality for only hash logic taken from https://github.com/alanshaw/ipfs-only-hash/blob/master/index.js
+
+export type Content = NodeJS.ReadableStream | Buffer | string
```

Comment:
will NodeJS.ReadableStream be available on the browser?

Reply:
i think in the case of fs above it's fine because libs handles that explicitly https://github.com/AudiusProject/audius-protocol/blob/master/libs/package.json#L8

Reply:
I used that type because line 136 (previously used in the JS implementation)

`if (content instanceof Stream.Readable) {`

and its `Stream` import are from node and implement NodeJS.ReadableStream. since the alternative is to change the way this converts to buffer (which is core to hashing non-image files so it's risky), I think we should explicitly disable using fileHasher in the browser (`"./utils/fileHasher": false`?) or use browserify -- what do you think?

Reply:
yeah you're porting a server side js file to a package that can be used client side so totally understand. i don't think browserify is viable given how heavy of a lift it is. i actually don't expect this to be used client side but the thing I was concerned with, like will a NodeJS type exist on the client or will that err? disabling makes a lot of sense though